[OSDOCS-7725] Shiftstack 4.14 release notes (#66243)
openshift/openshift-docs: release_notes/ocp-4-14-release-notes.adoc

==== Support for provisioning IPv6 spoke clusters from dual-stack hub clusters
With this update, you can provision IPv6 address spoke clusters from dual-stack hub clusters. In a zero touch provisioning (ZTP) environment, the HTTP server on the hub cluster that hosts the boot ISO now listens on both IPv4 and IPv6 networks. The provisioning service also checks the baseboard management controller (BMC) address scheme on the target spoke cluster and provides a matching URL for the installation media. These updates offer the ability to provision single-stack, IPv6 spoke clusters from a dual-stack hub cluster.

> [id="ocp-4-14-nw-shiftstack-dual-stack"]

Reviewer: It might be worth to point the user to the dual-stack docs for more information like it was done for CPMS release notes.
https://github.com/openshift/openshift-docs/pull/63935

Reviewer: BTW, the dual-stack docs gives additional context on what scenarios are supported and what are not.

==== Support for dual-stack networking for {rh-openstack} clusters (Technology Preview)

> Dual-stack network configuration is now available for clusters that run on {rh-openstack}. You can configure dual-stack networking during the deployment of a cluster on installer-provisioned infrastructure.
> 

Author: ```suggestion
Dual-stack network configuration is now available for clusters that run on {rh-openstack}. You can configure dual-stack networking during the deployment of a cluster on installer-provisioned infrastructure.

For more information, see xref:../installing/installing_openstack/installing-openstack-installer-custom.adoc#install-osp-dualstack_installing-openstack-installer-custom#install-osp-dualstack_installing-openstack-installer-custom[Configuring a cluster with dual-stack networking].
```

[id="ocp-4-14-nw-shiftstack-manage-security-groups"]

> ==== Security group management for {rh-openstack} clusters

Reviewer: might be worth to point to the feature docs, in case there is any

Author: There are some instructions in the installation assemblies, but I'm not sure how useful dropping someone partway into an installation process is.

In {product-title} 4.14, security for clusters that run on {rh-openstack} is enhanced. By default, the OpenStack cloud provider now sets the `manage-security-groups` option for load balancers to `true`, ensuring that only node ports that are required for cluster operation are open. Previously, security groups for both compute and control plane machines were configured to open a wide range of node ports for all incoming traffic.  

You can opt to use the previous configuration by setting the `manage-security-groups` option to `false` in the configuration of a load balancer. 

> For clusters that are upgraded to 4.14, you must manually update security rules that open the deployment to all traffic, i.e. permitting `0.0.0.0/0`.  

Reviewer: you mean users must manually update the sg rules to restrict more traffic?

Author: ```suggestion
For clusters that are upgraded to 4.14, you must manually update security group rules that open the deployment to all traffic, i.e. permitting `0.0.0.0/0`.  
```

==== Support for provisioning IPv6 spoke clusters from dual-stack hub clusters
With this update, you can provision IPv6 address spoke clusters from dual-stack hub clusters. In a zero touch provisioning (ZTP) environment, the HTTP server on the hub cluster that hosts the boot ISO now listens on both IPv4 and IPv6 networks. The provisioning service also checks the baseboard management controller (BMC) address scheme on the target spoke cluster and provides a matching URL for the installation media. These updates offer the ability to provision single-stack, IPv6 spoke clusters from a dual-stack hub cluster.

[id="ocp-4-14-nw-shiftstack-dual-stack"]
==== Support for dual-stack networking for {rh-openstack} clusters (Technology Preview)

Dual-stack network configuration is now available for clusters that run on {rh-openstack}. You can configure dual-stack networking during the deployment of a cluster on installer-provisioned infrastructure.

> For more information, see xref:../installing/installing_openstack/installing-openstack-installer-custom.adoc#install-osp-dualstack_installing-openstack-installer-custom#install-osp-dualstack_installing-openstack-installer-custom[Configuring a cluster with dual-stack networking].

Author: ```suggestion
For more information, see xref:../installing/installing_openstack/installing-openstack-installer-custom.adoc#install-osp-dualstack_installing-openstack-installer-custom[Configuring a cluster with dual-stack networking].

```

[id="ocp-4-14-nw-shiftstack-dual-stack"]
==== Support for dual-stack networking for {rh-openstack} clusters (Technology Preview)

Dual-stack network configuration is now available for clusters that run on {rh-openstack}. You can configure dual-stack networking during the deployment of a cluster on installer-provisioned infrastructure.

For more information, see xref:../installing/installing_openstack/installing-openstack-installer-custom.adoc#install-osp-dualstack_installing-openstack-installer-custom[Configuring a cluster with dual-stack networking].

[id="ocp-4-14-nw-shiftstack-manage-security-groups"]
==== Security group management for {rh-openstack} clusters

In {product-title} 4.14, security for clusters that run on {rh-openstack} is enhanced. By default, the OpenStack cloud provider now sets the `manage-security-groups` option for load balancers to `true`, ensuring that only node ports that are required for cluster operation are open. Previously, security groups for both compute and control plane machines were configured to open a wide range of node ports for all incoming traffic.  

> You can opt to use the previous configuration by setting the `manage-security-groups` option to `false` in the configuration of a load balancer. 

Reviewer 2: We should probably be more specific here, as setting `manage-security-groups` option to `false` alone won't work if you don't have the range open. Something along those lines:
```suggestion
You can opt to use the previous configuration by setting the `manage-security-groups` option to `false` in the configuration of a load balancer and ensuring that the security group rules to tne node ports range permit traffic from `0.0.0.0/0`.
```

Author: ```suggestion
You can opt to use the previous configuration by setting the `manage-security-groups` option to `false` in the configuration of a load balancer and ensuring that the security group rules to the node ports range permit traffic from `0.0.0.0/0`.
```

> For clusters that are upgraded to 4.14, you must manually update security group rules that open the deployment to all traffic, i.e. permitting `0.0.0.0/0`.  

Reviewer 2: Is this instruction to remove the permissive security group rules? It's not clear to me.

Author: @mandre How about:

```suggestion
For clusters that are upgraded to 4.14, you must manually remove permissive security group rules that open the deployment to all traffic. For example, you must remove a rule that permits traffic from `0.0.0.0/0`.  
```

Reviewer 2: Perfect :+1:

[id="ocp-4-14-nw-shiftstack-dual-stack"]
==== Support for dual-stack networking for {rh-openstack} clusters (Technology Preview)

Dual-stack network configuration is now available for clusters that run on {rh-openstack}. You can configure dual-stack networking during the deployment of a cluster on installer-provisioned infrastructure.

For more information, see xref:../installing/installing_openstack/installing-openstack-installer-custom.adoc#install-osp-dualstack_installing-openstack-installer-custom[Configuring a cluster with dual-stack networking].

[id="ocp-4-14-nw-shiftstack-manage-security-groups"]
==== Security group management for {rh-openstack} clusters

In {product-title} 4.14, security for clusters that run on {rh-openstack} is enhanced. By default, the OpenStack cloud provider now sets the `manage-security-groups` option for load balancers to `true`, ensuring that only node ports that are required for cluster operation are open. Previously, security groups for both compute and control plane machines were configured to open a wide range of node ports for all incoming traffic.

> You can opt to use the previous configuration by setting the `manage-security-groups` option to `false` in the configuration of a load balancer and ensuring that the security group rules to the node ports range permit traffic from `0.0.0.0/0`.

Author: ```suggestion
You can opt to use the previous configuration by setting the `manage-security-groups` option to `false` in the configuration of a load balancer and ensuring that the security group rules to the node ports range permit traffic from `0.0.0.0/0` on the node ports range 30000 through 32767.
```

Reviewer 2: Yes, let's just remove the repetition:
```suggestion
You can opt to use the previous configuration by setting the `manage-security-groups` option to `false` in the configuration of a load balancer and ensuring that the security group rules permit traffic from `0.0.0.0/0` on the node ports range 30000 through 32767.
```

Author: Whoops. Adding.

> For clusters that are upgraded to 4.14, you must manually remove permissive security group rules that open the deployment to all traffic. For example, you must remove a rule that permits traffic from `0.0.0.0/0`.

Reviewer 3: Uhm, maybe we can be more specific here - "traffic from 0.0.0.0/0 on node ports range (30000-32767)".

Author: ```suggestion
For clusters that are upgraded to 4.14, you must manually remove permissive security group rules that open the deployment to all traffic. For example, you must remove a rule that permits traffic from `0.0.0.0/0` on the node ports range 30000 through 32767.
```

@dulek

Reviewer 2: Yep, should we add the same to line 1192?

Author: https://github.com/openshift/openshift-docs/pull/66243#discussion_r1372010303

+
(link:https://issues.redhat.com/browse/OCPBUGS-15997[*OCPBUGS-15997*])

> * For clusters that run on {rh-openstack} 17.1 and use network function virtualization (NFV), a known issue in {rh-openstack} prevents succesful cluster deployment. There is no workaround for this issue. A hotfix that addresses the problem will be released soon.

Author: @MaysaMacedo Not sure what to put here for that OSP bug--is it likely that the hotfix will be released before OCP 4.14 GA?

Reviewer: @maxwelldb It's unlikely that the fix will be released before OCP GA. Perhaps the following could be mentioned instead: "While there is not fix released, customers can request a hotfix".

@eurijon what do you think?

Author: ```suggestion
* For clusters that run on {rh-openstack} 17.1 and use network function virtualization (NFV), a known issue in {rh-openstack} prevents succesful cluster deployment. There is no workaround for this issue. Contact Red Hat Support to request a hotfix.
```

Reviewer 4: The change looks good, the OSP fix won't be available before 4.14 GA.
cc @zivgreenberg

[cols="4,1,1,1",options="header"]
|====
|Feature |4.12 |4.13 |4.14

> |External load balancers with installer-provisioned infrastructure

Reviewer 3: Just double-checking - is this correct? I though we've added that in a previous release?

Reviewer: It's correct, external load-balancers is GA in 4.14

* Installation fails when installing {product-title} with the `bootMode` set to `UEFISecureBoot` on a node where Secure Boot is disabled. Subsequent attempts to install {product-title} with Secure Boot enabled will proceed normally. (link:https://issues.redhat.com/browse/OCPBUGS-19884)[*OCPBUGS-19884*])

> * For clusters that run on {rh-openstack} 17.1 and use network function virtualization (NFV), a known issue in {rh-openstack} prevents succesful cluster deployment. There is no workaround for this issue. Contact Red Hat Support to request a hotfix.

Reviewer 3: Can we update this with "or use Kuryr"? I think we opt out from Kuryr support with 17.1 given that it's going EOL in 4.15 and it's a significant testing cost. Nevertheless it does not work correctly at the moment due to Neutron issues and we can clearly state that.

Reviewer: @maxwelldb I think something like following can be more specific about the Kuryr scenario:

For clusters that run on {rh-openstack} 17.1 and use network function virtualization (NFV), a known issue in {rh-openstack} prevents successful cluster deployment. There is no workaround for this issue. Contact Red Hat Support to request a hotfix. Also, there is not support for Kuryr installations on  {rh-openstack} 17.1.

Author: Works for me. 

```suggestion
* For clusters that run on {rh-openstack} 17.1 and use network function virtualization (NFV), a known issue in {rh-openstack} prevents succesful cluster deployment. There is no workaround for this issue. Contact Red Hat Support to request a hotfix.r deployment. There is no workaround for this issue. Contact Red Hat Support to request a hotfix. Additionally, there is no support for Kuryr installations on {rh-openstack} 17.1.
```

* There is a known issue after enabling and disabling IPsec during runtime that causes the cluster to be in an unhealthy state with the error message: `an unknown error has occurred: MultipleErrors`. (link:https://issues.redhat.com/browse/OCPBUGS-19408[*OCPBUGS-19408*])

* Creating pods with Microsoft Azure File NFS volumes that are scheduled to the control plane node causes the mount to be denied.

> * For clusters that run on {rh-openstack} 17.1 and use network function virtualization (NFV), a known issue in {rh-openstack} prevents succesful cluster deployment. There is no workaround for this issue. Contact Red Hat Support to request a hotfix. Additionally, there is no support for Kuryr installations on {rh-openstack} 17.1.

Reviewer 4: I'm sorry if I missed a previous conversation about this, shouldn't we add the bz https://bugzilla.redhat.com/show_bug.cgi?id=2228643 here?

Author: ```suggestion
* For clusters that run on {rh-openstack} 17.1 and use network function virtualization (NFV), a known issue in {rh-openstack} prevents succesful cluster deployment. There is no workaround for this issue. Contact Red Hat Support to request a hotfix. Additionally, there is no support for Kuryr installations on {rh-openstack} 17.1. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2228643[*BZ2228643*])
```

+

> To work around this issue: If your control plane nodes are schedulable, and the pods can run on worker nodes, use `nodeSelector` or Affinity to schedule the pod in worker nodes. (link:https://issues.redhat.com/browse/OCPBUGS-18581[*OCPBUGS-18581*])

Reviewer 4: Looks like this workaround refers to the previous issue "Creating pods with Microsoft Azure File NFS volumes that are scheduled to the control plane node causes the mount to be denied.". In that case it should be placed just after the previous issue.

Author: Good catch. We spend just about every minute of the week up to freeze resolving merge conflicts--must be a mistake in that.

* Creating pods with Microsoft Azure File NFS volumes that are scheduled to the control plane node causes the mount to be denied.
+
To work around this issue: If your control plane nodes are schedulable, and the pods can run on worker nodes, use `nodeSelector` or Affinity to schedule the pod in worker nodes. (link:https://issues.redhat.com/browse/OCPBUGS-18581[*OCPBUGS-18581*])

> * For clusters that run on {rh-openstack} 17.1 and use network function virtualization (NFV), a known issue in {rh-openstack} prevents succesful cluster deployment. There is no workaround for this issue. Contact Red Hat Support to request a hotfix. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2228643[*BZ2228643*])

Reviewer 4: ```suggestion
* For clusters that run on {rh-openstack} 17.1 and use network function virtualization (NFV), a known issue in {rh-openstack} prevents successful cluster deployment. There is no workaround for this issue. Contact Red Hat Support to request a hotfix. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2228643[*BZ2228643*])
```

* Creating pods with Microsoft Azure File NFS volumes that are scheduled to the control plane node causes the mount to be denied.
+
To work around this issue: If your control plane nodes are schedulable, and the pods can run on worker nodes, use `nodeSelector` or Affinity to schedule the pod in worker nodes. (link:https://issues.redhat.com/browse/OCPBUGS-18581[*OCPBUGS-18581*])

> * For clusters that run on {rh-openstack} 17.1 and use network function virtualization (NFV), a known issue in {rh-openstack} prevents succesful cluster deployment. There is no workaround for this issue. Contact Red Hat Support to request a hotfix. Additionally, there is no support for Kuryr installations on {rh-openstack} 17.1.

Author: ```suggestion
* For clusters that run on {rh-openstack} 17.1 and use network function virtualization (NFV), a known issue in {rh-openstack} prevents succesful cluster deployment. There is no workaround for this issue. Contact Red Hat Support to request a hotfix. Additionally, there is no support for Kuryr installations on {rh-openstack} 17.1. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2228643[*BZ2228643*])
```

> * For clusters that run on {rh-openstack} 17.1 and use network function virtualization (NFV), a known issue in {rh-openstack} prevents succesful cluster deployment. There is no workaround for this issue. Contact Red Hat Support to request a hotfix. Additionally, there is no support for Kuryr installations on {rh-openstack} 17.1. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2228643[*BZ2228643*])

Reviewer 4: Sorry I forgot to mention that BZ2228643 belongs to the NFV issue, so my suggestion is:
* For clusters that run on {rh-openstack} 17.1 and use network function virtualization (NFV), a known issue in {rh-openstack} prevents succesful cluster deployment. There is no workaround for this issue. Contact Red Hat Support to request a hotfix. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2228643[*BZ2228643*])
* There is no support for Kuryr installations on {rh-openstack} 17.1.

I don't know if there was any reason not to add the "lack of Kuryr support in 17.1" on it's own known issue "bullet"

Author: ```suggestion
* For clusters that run on {rh-openstack} 17.1 and use network function virtualization (NFV), a known issue in {rh-openstack} prevents succesful cluster deployment. There is no workaround for this issue. Contact Red Hat Support to request a hotfix. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2228643[*BZ2228643*])

*  There is no support for Kuryr installations on {rh-openstack} 17.1. 
```

Author: @eurijon Done.
